packages/cw721/src/msg.rs

```diff
@@ -21,15 +47,288 @@ pub enum Cw721ExecuteMsg {
         expires: Option<Expiration>,
     },
     /// Remove previously granted Approval
-    Revoke { spender: String, token_id: String },
+    Revoke {
+        spender: String,
+        token_id: String,
+    },
     /// Allows operator to transfer / send any token from the owner's account.
     /// If expiration is set, then this allowance has a time/height limit
     ApproveAll {
         operator: String,
         expires: Option<Expiration>,
     },
     /// Remove previously granted ApproveAll permission
-    RevokeAll { operator: String },
+    RevokeAll {
+        operator: String,
+    },
+
+    /// Mint a new NFT, can only be called by the contract minter
+    Mint {
+        /// Unique ID of the NFT
+        token_id: String,
+        /// The owner of the newly minter NFT
+        owner: String,
+        /// Universal resource identifier for this NFT
+        /// Should point to a JSON file that conforms to the ERC721
+        /// Metadata JSON Schema
+        token_uri: Option<String>,
+        /// Any custom extension used by this contract
+        extension: TMetadataExtension,
+    },
+
     /// Burn an NFT the sender has access to
-    Burn { token_id: String },
+    Burn {
+        token_id: String,
+    },
+
+    /// Metadata msg
+    #[deprecated(since = "0.19.0", note = "Please use UpdateMetadata instead")]
+    /// Deprecated: use UpdateMetadata instead! In previous release it was a no-op for customization in other contracts. Will be removed in next release!
+    Extension {
+        msg: TMetadataExtensionMsg,
+    },
+    /// The creator is the only one eligible to update NFT onchain metadata (`NftInfo.extension`).
+    UpdateMetadata {
+        token_id: String,
+        extension: TMetadataExtensionMsg,
+    },
+
+    /// Sets address to send withdrawn fees to. Only owner can call this.
+    SetWithdrawAddress {
+        address: String,
+    },
+    /// Removes the withdraw address, so fees are sent to the contract. Only owner can call this.
+    RemoveWithdrawAddress {},
+    /// Withdraw from the contract to the given address. Anyone can call this,
+    /// which is okay since withdraw address has been set by owner.
+    WithdrawFunds {
+        amount: Coin,
+    },
+}
+
+#[cw_serde]
+pub struct Cw721InstantiateMsg<TCollectionInfoExtension> {
+    /// Name of the NFT contract
+    pub name: String,
+    /// Symbol of the NFT contract
+    pub symbol: String,
+
+    pub collection_info_extension: TCollectionInfoExtension,
+
+    /// The minter is the only one who can create new NFTs.
+    /// This is designed for a base NFT that is controlled by an external program
+    /// or contract. You will likely replace this with custom logic in custom NFTs
+    pub minter: Option<String>,
+
+    /// Sets the creator of collection. The creator is the only one eligible to update `CollectionInfo`.
+    pub creator: Option<String>,
```

Comment:
@JakeHartnell added more in docs explaining what creator can do.

Reply:
Thanks!